src/datafactory/azext_datafactory/generated/_params.py

```diff
@@ -0,0 +1,633 @@
+from azext_datafactory.action import (
+    AddFactoryVstsConfiguration,
+    AddFactoryGitHubConfiguration,
+    AddFilters,
+    AddOrderBy,
+    AddDebugSettingsSourceSettings,
+    AddCommandPayload
+)
+
+
+def load_arguments(self, _):
+
+    with self.argument_context('datafactory factory list') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+
+    with self.argument_context('datafactory factory show') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('factory_name', options_list=['--name', '-n'], help='The factory name.', id_part='name')
+        c.argument('if_none_match', help='ETag of the factory entity. Should only be specified for get. If the ETag mat'
+                   'ches the existing entity tag, or if * was provided, then no content will be returned.')
+
+    with self.argument_context('datafactory factory create') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('factory_name', options_list=['--name', '-n'], help='The factory name.')
+        c.argument('if_match',
+                   help='ETag of the factory entity. Should only be specified for update, for which it shou'
+                        'ld match existing entity or can be * for unconditional update.')
+        c.argument('location',
+                   arg_type=get_location_type(self.cli_ctx),
+                   validator=get_default_location_from_resource_group)
+        c.argument('tags', tags_type)
+        c.argument('factory_vsts_configuration', action=AddFactoryVstsConfiguration, nargs='+', help='Factory\'s VSTS r'
+                   'epo information.', arg_group='RepoConfiguration')
+        c.argument('factory_git_hub_configuration', action=AddFactoryGitHubConfiguration, nargs='+', help='Factory\'s G'
+                   'itHub repo information.', arg_group='RepoConfiguration')
+        c.argument('global_parameters',
+                   arg_type=CLIArgumentType(options_list=['--global-parameters'],
+                                            help='List of parameters for factory. Expected value: '
+                                                 'json-string/@json-file.'))
+
+    with self.argument_context('datafactory factory update') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('factory_name', options_list=['--name', '-n'], help='The factory name.', id_part='name')
+        c.argument('tags', tags_type)
+
+    with self.argument_context('datafactory factory delete') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('factory_name', options_list=['--name', '-n'], help='The factory name.', id_part='name')
+
+    with self.argument_context('datafactory factory configure-factory-repo') as c:
+        c.argument('location_id', help='The location identifier.', id_part='name')

```

Comment:
Can we name the option to be `--location` and use `get_location_type`?

Reply:
This should have changed now. please review again. Thx